lf/demo/demo_robot.lf

```diff
@@ -2,24 +2,53 @@ target C {
     timeout: 15 sec
 };
 
-reactor Sensors {    
+// This is what we call a driver container. But I think at Luos we will have 2 drivers, one for the button and one for bumper.
+// But perhaps its more lik nodes.
+reactor Sensors {
+    // Those are like pipe beetween reactors?
     output button_out:bool;
     output bumper_out:bool;
     
+    // State are like shared variables?
+
+    // In Luos basic functionalities are represented by models (we call it template sometimes).
+    // For example "State" model is for buttons, and all the values representing the button can be output, input or both. They are associated to already maid "pipe" depending on the format of data (bool, distance, temperature, ...)
+    // To say it differently the "State" type we have already have the "output button_out:bool;" on it. And its a configurable option in this particular type.
+    // This kind of model is related to a container managing all the "pipes" of the model and sharing it with other containers.
+    // From another container, to get the value of a Button container I have to ask him a bool (we can also ask for all output data).
+    // Here is a Luos button example using models (template) : https://github.com/Luos-io/Examples/tree/template/Drivers/Button
+
+    // Users can create their hown models or don't use them and manage messages ("pipes") directly on the container code.
+    // Regarding your code we are describing variables and pipes like we make it when we don't use pre-made models (templates).
+    // To create a container like this Luos need some informations : 
+    // - Callback => A callback called when the container receive something. In this function we have the things you make in "reactions".
+    // - Type     => A kind of model ID. Some are already defined (motor, distance, ...), and users can create their own. I think it is like your "bool" for button.
```

Comment:
For the button example, is the type `STATE_MOD`? What does it mean?

Reply:
Yes the type is `STATE_MOD' wich is basicaly a boolean. Perhaps it doesn't mean anything for english people it could be a language mistake from our side ...